### PR TITLE
VIH-11017 Dynatrace Real User Monitoring

### PR DESCRIPTION
**Pull request by marcogagliardi**
### Jira link

[VIH-11017](https://tools.hmcts.net/jira/browse/VIH-11017)

### Change description

added RUM javascript tag

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

**Review by will-craig — COMMENTED** [review]
I think best practice is to add it to the scripts array in the angular.json

**Review by will-craig — COMMENTED** [review]
It looks good from what I can see, is there any way we can test this before merging

File: VideoWeb/VideoWeb/ClientApp/src/app/app.component.ts [hunk 1]
```diff
@@ -94,13 +96,23 @@ export class AppComponent implements OnInit, OnDestroy {
...
                             this.eventhubService.configureConnection();
+
+                            /* The line
+                            `this.dynatraceService.addUserIdentifyScript(userData?.preferred_username?.toLowerCase());`
+                            is calling a method `addUserIdentifyScript` from the `dynatraceService`
+                            service. This method is used to identify the user in Dynatrace
+                            monitoring by passing the user's preferred username in lowercase as a
+                            parameter.*/
+                            // TODO: uncomment the below line to add this identification
+                            // this.dynatraceService.addUserIdentifyScript(userData?.preferred_username?.toLowerCase());
```
Comment (will-craig) [inline VideoWeb/VideoWeb/ClientApp/src/app/app.component.ts:115]: Can we not test this yet?

Comment (marcogagliardi) [reply VideoWeb/VideoWeb/ClientApp/src/app/app.component.ts:115]: I can't understand. Sorry

Comment (will-craig) [reply VideoWeb/VideoWeb/ClientApp/src/app/app.component.ts:117]: If this is dependant on something else, can you reference the ticket number in the commented out code.

Comment (marcogagliardi) [reply VideoWeb/VideoWeb/ClientApp/src/app/app.component.ts:117]: done

File: VideoWeb/VideoWeb/ClientApp/src/app/services/api/dynatrace.service.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+     * The function `addUserIdentifyScript` creates a script element to identify a user with a given
+     * ID.
+     * @param userIdendify - The `userIdendify` parameter is the user identifier that will be passed to
+     * the `dtrum.identifyUser` function in the `addUserIdentifyScript` method. This identifier is used
+     * to uniquely identify a user within the Dynatrace monitoring system.
+     */
+    addUserIdentifyScript(userIdendify) {
+        const script = this.renderer.createElement('script');
+        script.text = 'dtrum.identifyUser("' + userIdendify + '")';
+        this.document.head.appendChild(script);
```
Comment (will-craig) [inline VideoWeb/VideoWeb/ClientApp/src/app/services/api/dynatrace.service.ts:39]: Just noticed this doesn't have any code coverage, could you add a quick unit test for this function too, please?

Comment (marcogagliardi) [reply VideoWeb/VideoWeb/ClientApp/src/app/services/api/dynatrace.service.ts:39]: done

Comment (shaed-parkar) [reply VideoWeb/VideoWeb/ClientApp/src/app/services/api/dynatrace.service.ts:39]: is `userIdendify` a typo? Should it be `userIdentity` ?

Comment (marcogagliardi) [reply VideoWeb/VideoWeb/ClientApp/src/app/services/api/dynatrace.service.ts:39]: no it is not a typo. It is injecting a script to call a method called identifyUser

**Review by will-craig — APPROVED** [review]

File: VideoWeb/VideoWeb/ClientApp/src/app/app.component.ts [hunk 3]
```diff
@@ -94,13 +96,25 @@ export class AppComponent implements OnInit, OnDestroy {
...
                             this.eventhubService.configureConnection();
+
+                            /* The line
+                            `this.dynatraceService.addUserIdentifyScript(userData?.preferred_username?.toLowerCase());`
+                            is calling a method `addUserIdentifyScript` from the `dynatraceService`
+                            service. This method is used to identify the user in Dynatrace
+                            monitoring by passing the user's preferred username in lowercase as a
+                            parameter.*/
+                            // TODO: uncomment the below line to add this identification
+                            // To be implemented behind the cookie banner VIH-5622
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply VideoWeb/VideoWeb/ClientApp/src/app/app.component.ts:115]: Michael is aware of this. He told me to link to this ticket

**Review by komaldabhi1509 — APPROVED** [review]

**Review by will-craig — APPROVED** [review]

**Pull request merged**